### PR TITLE
Integrate with the Mac Now Playing Info Center

### PR DESCRIPTION
**Pull request by dphoyes**
This adds MacOS support for controlling cantata via hardware media keys,
or via the OS's Now Playing widget. Implements #614 and #1347.

### CODE REVIEW

File: mac/macnowplaying.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Cantata
+ *
+ * Copyright (c) 2011-2021 Craig Drummond <craig.p.drummond@gmail.com>
```
Comment (Thaodan) [inline mac/macnowplaying.h:4]: The copyright is wrong it should be your name there.

File: mac/macnowplaying.mm [hunk 2]
```diff
@@ -0,0 +1,172 @@
+/*
+ * Cantata
+ *
+ * Copyright (c) 2011-2021 Craig Drummond <craig.p.drummond@gmail.com>
+ *
```
Comment (Thaodan) [inline mac/macnowplaying.mm:5]: Same again.

File: mac/macnowplaying.mm [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    impl()
+        : nowPlayingInfo([[NSMutableDictionary alloc] init])
+    {}
+};
+
+MacNowPlaying::MacNowPlaying(QObject *p)
+    : QObject(p)
+    , pimpl(new impl())
+{
+    connect(this, SIGNAL(setSeekId(qint32, quint32)), MPDConnection::self(), SLOT(setSeekId(qint32, quint32)));
```
Comment (Thaodan) [inline mac/macnowplaying.mm:50]: I don't think Qt4 stlye connect should be used anymore.
See https://doc.qt.io/qt-5/signalsandslots.html

File: CMakeLists.txt [hunk 4]
```diff
@@ -281,7 +281,14 @@ if (APPLE)
         set(CANTATA_SRCS ${CANTATA_SRCS} mac/powermanagement.cpp)
     endif ()
     find_package(Foundation REQUIRED)
-    set(CANTATA_LIBS ${CANTATA_LIBS} ${FOUNDATION_LIBRARY})
+    find_library(MAC_APPKIT_LIBRARY AppKit REQUIRED)
```
Comment (Thaodan) [inline CMakeLists.txt:284]: Should this not be optional to not break older releases?

**Review by Thaodan — CHANGES_REQUESTED** [review]
See the attached comments.

File: mac/macnowplaying.h (hunk 1 above)
Comment (CDrummond) [reply mac/macnowplaying.h:4]: I agree, this file is yours so feel free to add your name.

File: CMakeLists.txt (hunk 4 above)
Comment (CDrummond) [reply CMakeLists.txt:284]: I don't see why - I'd imagine AppKit is standard on a Mac.

**Comment by CDrummond** [comment]
Thanks for this. When you've updated the copyright I'll happily merge.

File: mac/macnowplaying.h (hunk 1 above)
Comment (dphoyes) [reply mac/macnowplaying.h]: Thanks, updated.

File: CMakeLists.txt (hunk 4 above)
Comment (dphoyes) [reply CMakeLists.txt:284]: https://developer.apple.com/documentation/appkit?language=objc says that AppKit is available in macOS 10.0+, which is the same as for Foundation.

**Pull request merged**